ibis/backends/base/sql/registry/window.py

```diff
@@ -54,36 +55,22 @@ def _replace_interval_with_scalar(expr: Union[ir.Expr, dt.Interval, float]):
     -------
     preceding : float or ir.FloatingScalar, depending upon the expr
     """
-    if isinstance(expr, ir.Expr):
-        expr_op = expr.op()
-    else:
-        expr_op = None
-
-    if not isinstance(expr, (dt.Interval, ir.IntervalValue)):
-        # Literal expressions have op method but native types do not.
-        if isinstance(expr_op, ops.Literal):
-            return expr_op.value
-        else:
-            return expr
-    elif isinstance(expr, dt.Interval):
+    if isinstance(op, ops.Literal):
+        unit = getattr(op.output_dtype, "unit", "us")
```

Comment:
During the refactors my general experience was that `getattr` this way made it harder to locate type problems by swallowing attribute errors. 
Based on the function's name `_replace_interval_with_scalar` I'd expect to always have an interval dtype here, are other inputs allowed?

Reply:
Other inputs are allowed, e.g. in `2 * my_interval` the recursion will eventually hit the `2` which will have an `int8` dtype.